File: OneCodeTeam/How-to-log-UI-traces-into-Windows-Azure-table-29443/[VB]-How to log UI traces into Windows Azure table storage/VB/VBAzureJSLogging/Controllers/HomeController.vb
'***************************** Module Header ******************************\
' Module Name: HomeController.vb
' Project:     VBAzureJSLogging
' Copyright (c) Microsoft Corporation
'
' HomeController
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'****************************************************************************/
Imports System.Web.Mvc

Namespace Controllers
    Public Class HomeController
        Inherits Controller

        ' GET: Home
        Function Index() As ActionResult
            Return View()
        End Function
    End Class
End Namespace

File: OneCodeTeam/How-to-construct-data-to-json-string-before-27410/[C#]-How to construct data to json string before posting to pagemethod using jQuery/C#/CSASPNETUseJsonString/Controllers/HomeController.cs
/***************************** Module Header ******************************\
* Module Name:	HomeController.cs
* Project:		CSASPNETUseJsonString
* Copyright (c) Microsoft Corporation.
* 
* This sample shows how to post complex data to MVC server side using json string.
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\**************************************************************************/

using CSASPNETUseJsonString.Models;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace CSASPNETUseJsonString.Controllers
{
    public class HomeController : Controller
    {
        //
        // GET: /Home/

        public ActionResult Index()
        {
            return View();
        }

        [HttpPost]
        public ActionResult saveUserInfos(string data)
        {
            try
            {
                var userInfoList = JsonConvert.DeserializeObject<IEnumerable<UserInfo>>(data);
                // Save these data to Your DB

                return Json("Data successfully saved.");
            }
            catch (JsonReaderException e)
            {

                return Json(e.Message);
            }
        }

    }
}
